### PR TITLE
Add mesh utilities softwares

### PR DESCRIPTION
**Pull request by gregoire-dl**
## Description

Add two softwares based on Geogram for mesh boolean operations and mesh conversion.

## Features list

- [x] Add `MergeMeshes` software
- [x] Add `ConvertMesh` software

### CODE REVIEW

File: src/software/utils/main_mergeMeshes.cpp [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    std::string verboseLevel = system::EVerboseLevel_enumToString(system::Logger::getDefaultVerboseLevel());
+    std::string inputFirstMeshPath;
+    std::string inputSecondMeshPath;
+    std::string outputFilePath;
+    std::string operationTypeName = EOperationType_enumToString(EOperationType::BOOLEAN_UNION);
+
+    bool preProcess = true;
+    bool postProcess = true;
+
+    po::options_description allParams("AliceVision mergeMeshes");
```
Comment (simogasp) [inline src/software/utils/main_mergeMeshes.cpp:138]: Please specify better what the program does e.g. "The program takes two meshes and..."

File: src/software/convert/main_convertMesh.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    system::Timer timer;
+
+    // command-line parameters
+
+    std::string verboseLevel = system::EVerboseLevel_enumToString(system::Logger::getDefaultVerboseLevel());
+    std::string inputMeshPath;
+    std::string outputFilePath;
+
+    po::options_description allParams("AliceVision convertMesh");
```
Comment (simogasp) [inline src/software/convert/main_convertMesh.cpp:51]: same here

**Review by simogasp — CHANGES_REQUESTED** [review]

File: src/software/convert/main_convertMesh.cpp [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        if(!outputFolderPart.empty() && !fs::exists(outputFolderPart))
+        {
+            if(!fs::create_directory(outputFolderPart))
+            {
+                ALICEVISION_LOG_ERROR("Cannot create output folder");
+                return EXIT_FAILURE;
+            }
+        }
+    }
```
Comment (simogasp) [inline src/software/convert/main_convertMesh.cpp:126]: maybe we can refactor in function inside alicevision_system to be used in all the other exe

File: src/software/utils/main_mergeMeshes.cpp [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+    GEO::initialize();
+    GEO::CmdLine::import_arg_group("standard");
+    GEO::CmdLine::import_arg_group("algo");
+
+    ALICEVISION_LOG_INFO("Geogram initialized.");
+
+    GEO::Mesh inputFirstMesh, inputSecondMesh, outputMesh;
+
+    {
```
Comment (simogasp) [inline src/software/utils/main_mergeMeshes.cpp:242]: block not really necessary?

File: src/software/utils/main_mergeMeshes.cpp (hunk 1 above)
Comment (gregoire-dl) [reply src/software/utils/main_mergeMeshes.cpp]: done, thanks

File: src/software/convert/main_convertMesh.cpp (hunk 2 above)
Comment (gregoire-dl) [reply src/software/convert/main_convertMesh.cpp]: done, thanks

File: src/software/utils/main_mergeMeshes.cpp (hunk 4 above)
Comment (gregoire-dl) [reply src/software/utils/main_mergeMeshes.cpp:246]: yes ! Done, thanks

File: src/software/convert/main_convertMesh.cpp (hunk 3 above)
Comment (gregoire-dl) [reply src/software/convert/main_convertMesh.cpp:127]: It's a good idea. But maybe it's better to do that in another PR with a larger scope ?

Comment (simogasp) [reply src/software/convert/main_convertMesh.cpp:127]: we can start with this one creating the function in the library and replacing the code in these 2.

I don't know how many other blocks are there like this one in the other binaries but it's a matter of search and replace

**Review by fabiencastan — APPROVED** [review]

**Review by fabiencastan — APPROVED** [review]

File: src/software/utils/main_mergeMeshes.cpp [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+/**
+ * @brief get informations about each operation type
+ * @return String
+ */
+std::string EOperationType_informations()
+{
+    return "Operation types used to merge two meshes:\n"
+           "* boolean_union: Create a new mesh with the combined volume of the two input meshes.\n"
+           "* boolean_intersection: Create a new mesh from the intersected volumes of the two input meshes.\n"
+           "* boolean_difference: Create a new mesh from the volume of the first input mesh substracted by the second input mesh.\n";
```
Comment (natowi) [inline src/software/utils/main_mergeMeshes.cpp:56]: Typo: _subtracted_ not _substracted_

**Pull request merged**